test/e2e/specs/site-editor/block-removal.spec.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Site editor block removal prompt', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test.beforeEach( async ( { admin, editor } ) => {
+		await admin.visitSiteEditor( {
+			postId: 'emptytheme//index',
+			postType: 'wp_template',
+		} );
+		await editor.canvas.click( 'body' );
+	} );
+
+	test( 'should appear when attempting to remove Query Block', async ( {
+		page,
+	} ) => {
+		// Open and focus List View
+		await page.getByRole( 'button', { name: 'List View' } ).click();
```

Comment:
Suggestion (non-blocking): It might be a good idea to scope the selector better here. This could prevent future flaky tests like #51636 and #51564.

Reply:
Great tip, thanks